python/pyspark/sql/functions.py

```diff
@@ -2382,6 +2382,20 @@ def array_sort(col):
     return Column(sc._jvm.functions.array_sort(_to_java_column(col)))
 
 
+@since(2.4)
+def shuffle(col):
+    """
+    Collection function: Generates a random permutation of the given array.
+
+    .. note:: The function is non-deterministic because its results depends on order of rows which
```

Comment:
Maybe this one would be better? "The function is non-deterministic because it produces 
    an unbiased permutation: every permutation is equally likely."

Reply:
The permutation is determined randomly but it is determined for the same query plan if the order of rows is determined, because the analyzer will assign a random seed for it.

Reply:
given a same input sequence, will this function always return the same permutation?

Reply:
The seed is fixed when analysis phase, so if we, say, `collect()` twice or more from the same DataFrame, we will get the same result:

```scala
val df = .. .select(shuffle('arr))
df.collect() == df.collect()
```

but if we create another DataFrame from the same input, we will get different results:

```scala
val df1 = .. .select(shuffle('arr))
val df2 = .. .select(shuffle('arr))
df1.collect() != df2.collect()
```